services/worker-manager/src/provider_google.js

```
@@ -22,86 +65,502 @@ class GoogleProvider extends Provider {
+          zone,
+          requestId: uuid.v4(), // This is just for idempotency
+          requestBody: {
+            name: instanceName,
+            labels: {
+              workertype: workerType.name,
+            },
+            description: workerType.description,
+            machineType: `zones/${zone}/machineTypes/${workerType.config.machineType}`,
+            scheduling: workerType.config.scheduling,
+            networkInterfaces: workerType.config.networkInterfaces,
+            disks: workerType.config.disks,
+            serviceAccounts: [{
+              email: this.workerAccountEmail,
+              scopes: [
+                /*
+                 * This looks scary but is ok. According to
+                 * https://cloud.google.com/compute/docs/access/service-accounts#accesscopesiam
+                 *
+                 * "A best practice is to set the full cloud-platform
+                 * access scope on the instance, then securely limit
+                 * the service account's API access with IAM roles."
+                 *
+                 * Which is what we do.
+                 */
+                'https://www.googleapis.com/auth/cloud-platform',
+              ],
+            }],
+            metadata: {
+              items: [
+                {
+                  key: 'taskcluster',
+                  value: JSON.stringify({
+                    provisionerId: this.provisionerId,
+                    workerType: workerType.name,
+                    workerGroup: `${workerType.name}-google`,
+                    credentialUrl: libUrls.api(this.rootUrl, 'worker-manager', 'v1', `credentials/google/${workerType.name}`),
+                    rootUrl: this.rootUrl,
+                    userData: workerType.config.userData,
+                  }),
+                },
+              ],
+            },
+          },
+        });
+        op = res.data;
+      } catch (err) {
+        if (!err.errors) {
+          throw err;
```

do you want to report this error too?